c2rust-analyze/src/main.rs

```diff
@@ -482,6 +500,24 @@ fn run(tcx: TyCtxt) {
         }
     }
 
+    // FIX the fields of structs mentioned in extern blocks
+    for adt_did in &gacx.adt_metadata.struct_dids {
+        let ty = &tcx.type_of(adt_did);
+        if gacx.foreign_mentioned_tys.contains(ty) {
+            if let TyKind::Adt(adt_def, _) = tcx.type_of(adt_did).kind() {
```

Comment:
Are any other types expected to be supported?

Reply:
Yeah, so this suggestion doesn't really apply/is useful anymore.  See https://github.com/immunant/c2rust/pull/960/files/b90b6465c179fb8b5913d3b09fa5c17b220c3c1d#r1230230044 for which types to recurse into.

Reply:
There are other comments outlining the need for recursive visitation, so closing this comment and will close the others once recursive visitation is supported.

Reply:
So AFAIK ADTs are the only types that we recurse into that we want to avoid rewriting explicitly. I believe the rest are at least recursed into in 795438ae

Reply:
Ah, okay.  That makes sense, as ADTs are all the nominal (so not aliases) aggregate types that we'd rewrite.  In that case, you can use `tcx.adt_def` here again if it applies.

Reply:
addressed in f15d8041